File: Java/src/com/dbc/algorithm/Array/MaximumAverageSubarrayI.java
package com.dbc.algorithm.Array;

public class MaximumAverageSubarrayI {
    public static double findMaxAverage(int[] nums, int k) {
        int result = 0;
        for (int i = 0; i < k; i++) {
            result += nums[i];
        }
        int remain = result;
        for (int i = k; i < nums.length; i++) {
            remain = remain + nums[i] - nums[i - k];
            result = Math.max(result, remain);
        }

        return (double) result / k;
    }
}

File: Python/array/maximum_average_subarray_i.py
from typing import List

def findMaxAverage(nums: List[int], k: int) -> float:
    result = sum(nums[:k])
    remain = sum(nums[:k])
    for i in range(k, len(nums)):
        remain = remain + nums[i] - nums[i - k]
        result = max(result, remain)
    
    return result / k

if __name__ == "__main__":
    print(findMaxAverage(nums=[0,4,0,3,2], k = 1))
